### ISSUE TITLE
Add support for milight

### ISSUE THREAD

**Issue by sideshowbob84** [tvdzwan/hyperion#390]
Hey,
i have some milight bulbs (cheap version of hue). It would be great to use these in hyperion. The API is open and there many projects on github. For example here is some c code doing the rgb conversion and sending the values via UDP. 
https://github.com/netdata/loxone/blob/master/applamp/milight.c

**Comment by r3vile** [comment]
Yeah milight support would be great. Maybe i could help testing and/or writing the leddevice class, but i would need some inital help there.

**Comment by rubberduck2006** [comment]
MiLight support would be nice. It's a proprietary protocol and should be easy to install. Hue is indeed supported ... equal rights for all;-) The following HomeLights have the same protocol (not exhaustive): 
• MiLight 
• EasyBulb 
• Limitless LED 
• iLight 
• AppLight 
• AppLamp 
• ledME 
For a workaround, I would be really grateful

**Comment by b2un0** [comment]
https://github.com/tvdzwan/hyperion/pull/444

**Comment by deennoo** [comment]
Nice !

How to setup for milight please ?

**Comment by rubberduck2006** [comment]
I think there's the protocol needs to be extended under libsrc / leddevice / LedDeviceUdp.cpp.
See here: http://www.limitlessled.com/dev/
At the moment I have no idea how to go to the, unfortunately.

**Comment by penfold42** [comment]
Correct - the current code supports multiple protocol types - 0,1 and 2 are used at the moment.

I won't be able to look at this for at least a week and I don't have a lamp to test with.

**Comment by b2un0** [comment]
if you have some examples to implement this i would give it a try - i have some milight bulbs with a wifi controller and would also control there with hyperion.

**Comment by deennoo** [comment]
Got some Milight a wifi bridge working to, my hyperion run on a Pi, can be your beta tester !

**Comment by deennoo** [comment]
I know this not the better way to work but look at this : 

http://epocapp.bitbucket.org/milight/

Epocapp made a plug in for kodi and windows to drive Milight

**Comment by tpmodding** [comment]
guys, have anyone a milight to test with the udp support?
@sideshowbob84 @deennoo @b2un0 @rubberduck2006 @penfold42

**Comment by b2un0** [comment]
@tpmodding yes, i have 5 milight devices and the wifi controller, but i have no idea how to test it with hyperion :(

@brindosch is it possible to add a LedDeviceMilight with some test code based on this python code?
https://github.com/McSwindler/python-milight

**Comment by tpmodding** [comment]
i can only guess...

but i think if we use the forwarder and use an udp-device, and enter the configs here ->
http://epocapp.bitbucket.org/milight/assets/img/help_settings.jpg

my it should work? dont know...

**Comment by brindosch** [comment]
The documentation is there
http://www.limitlessled.com/dev/
All values needs to be converted following this docu.

But there is one big downside of milight.
No saturation possible! Which is not that good for an ambilight. You will
never reach a philipshue
Am 22.04.2016 1:31 nachm. schrieb "tpmodding" <notifications@github.com>:

**Comment by TobiMiller** [comment]
would be awesome! :)

**Comment by galip91** [comment]
would love for this to happen :)

**Comment by EdgardosReis** [comment]
It would be awesome! And now milight supports saturation in their RGC+CCT bulbs.

**Comment by HardBug25** [comment]
Would love to see this! Any development on this?

**Comment by guustflater** [comment]
Please add milight support, that would be awesome!

**Comment by tpmodding** [comment]
not here anymore...please make an issue in hyperion.ng
